src/api/app/helpers/validation_helper.rb

```diff
@@ -10,45 +10,34 @@ def valid_package_name!(package_name)
 
   # load last package meta file and just check if sourceaccess flag was used at all, no per user checking atm
   def validate_read_access_of_deleted_package(project, name)
-    prj = Project.get_by_name(project)
-    if prj.is_a?(Project)
-      raise Project::ReadAccessError, project.to_s if prj.disabled_for?('access', nil, nil)
-      raise Package::ReadSourceAccessError, "#{target_project_name}/#{target_package_name}" if prj.disabled_for?('sourceaccess', nil, nil)
+    project_object = Project.find_by(name: project)
+    if project_object
+      raise Package::ReadSourceAccessError, "#{project}/#{name}" if project_object.disabled_for?('sourceaccess', nil, nil)
+    else
+      validate_visibility_of_deleted_project(project)
     end
 
     begin
-      revisions_list = Backend::Api::Sources::Package.revisions(project, name)
-    rescue StandardError
+      meta = Xmlhash.parse(Backend::Api::Sources::Package.meta(project, name, { deleted: 1 }))
+    rescue Backend::NotFoundError
       raise Package::UnknownObjectError, "Package not found: #{project}/#{name}"
     end
-    data = Xmlhash.parse(revisions_list)
-    lastrev = data.elements('revision').last
-
-    query = { deleted: 1 }
-    query[:rev] = lastrev.value('srcmd5') if lastrev
-    meta = PackageMetaFile.new(project_name: project, package_name: name).content(query)
-    raise Package::UnknownObjectError, "Package not found: #{project}/#{name}" unless meta
 
     return true if User.admin_session?
-    raise Package::ReadSourceAccessError, "#{project}/#{name}" if FlagHelper.xml_disabled_for?(Xmlhash.parse(meta), 'sourceaccess')
+    raise Package::ReadSourceAccessError, "#{project}/#{name}" if FlagHelper.xml_disabled_for?(meta, 'sourceaccess')
 
     true
   end
 
   def validate_visibility_of_deleted_project(project)
```

Comment:
This either returns `true`, `nil` or raises an error. What if it returns the `meta`? That way you can reuse it above in line 27...

Reply:
Not sure what you want to reuse? The project meta? For what?

Reply:
Ah nothing, I overlooked that they are different `meta`'s. Nvm